peers/app_request_network.go

```diff
@@ -42,7 +46,8 @@ type AppRequestNetwork struct {
 func NewNetwork(
 	logLevel logging.Level,
 	registerer prometheus.Registerer,
-	cfg *config.Config,
+	trackedSubnets set.Set[ids.ID],
```

Comment:
Yeah we definitely need to keep it for now and I want to save bumping different versions of all our dependencies for a later PR. I am sort of considering keeping it for now as well though since there are other differences/places where `trackedSubnets` are used in the networking code. 

Specifically uptime tracking via ping messages. Not having `trackedSubnets` set makes peer nodes spew more logs about uptime tracking. This uptime tracking via ping messages  should go away soon as well but it hasn't yet.

Reply:
Sounds good. A more general comment, there's a lot of miscellania that's rightly being deferred in this PR. Do you mind enumerating them in a followup ticket so that they don't get lost in the chaff?

Reply:
Yup I will enter those issues and link them in the description for easy reference before we are ok to merge this one.